File: arena-xunit/src/Lifecycle/DependencyState.cs
using System.Collections.Generic;

namespace ArenaDotnet.Xunit.Lifecycle;

public sealed class DependencyState
{
    public string Id { get; set; } = "";
    public string State { get; set; } = "";
    public List<Fault> Faults { get; set; } = new();
    public List<DependencyState> Children { get; set; } = new();

    public DependencyState? Find(string identifier)
    {
        if (Id == identifier)
            return this;
        foreach (var child in Children)
        {
            var found = child.Find(identifier);
            if (found != null)
                return found;
        }
        return null;
    }
}

File: arena-junit/src/main/java/arena/junit/lifecycle/DependencyState.java
package arena.junit.lifecycle;

import java.util.List;

public final class DependencyState {
  public String id = "";
  public String state = "";
  public List<Fault> faults = List.of();
  public List<DependencyState> children = List.of();

  public DependencyState find(String identifier) {
    if (id.equals(identifier)) {
      return this;
    }
    for (DependencyState child : children) {
      DependencyState found = child.find(identifier);
      if (found != null) {
        return found;
      }
    }
    return null;
  }
}
